fuel-tx/src/transaction/fee.rs

```diff
@@ -74,8 +93,10 @@ impl TransactionFee {
 
         // TODO: use native div_ceil once stabilized out from nightly
         let bytes_gas = params.gas_per_byte.checked_mul(metered_bytes)?;
-        let min_gas = bytes_gas.checked_add(gas_used_by_predicates)?;
-        let max_gas = bytes_gas.checked_add(gas_limit)?;
+        let min_gas = bytes_gas
+            .checked_add(gas_used_by_signature_checks)?
+            .checked_add(gas_used_by_predicates)?;
+        let max_gas = min_gas.checked_add(gas_limit)?;
```

Comment:
Fee is effectively calculated using [the formula defined in the spec](https://github.com/FuelLabs/fuel-specs/blob/master/src/protocol/tx-validity.md#script-execution). Min fee and max fee are separated by the gas limit. 

In effect, `max_fee = min_fee + gas_limit * gas_price`.